code-scan.js

```diff
@@ -0,0 +1,203 @@
+const { Octokit } = require("@octokit/rest");
+const core = require("@actions/core");
+const { countReset } = require("console");
+const fs = require("fs");
+const token = process.argv[2];
+const repo = process.argv[3];
+const originalOwner = process.argv[4];
+const owner = "asml-actions-validation";
+console.log(`repo name is ${repo}, original owner ${originalOwner}`);
+const octokit = new Octokit({
+  auth: token,
+});
+const octokitFunctions = {
+  getRepo: octokit.repos.get,
+  delRepo: octokit.repos.delete,
+  createFork: octokit.repos.createFork,
+  enableDependabot: octokit.rest.repos.enableVulnerabilityAlerts,
+  listAlertsForRepo: octokit.rest.dependabot.listAlertsForRepo,
+  listScanningResult: octokit.rest.codeScanning.listAlertsForRepo,
+  listLanguages: octokit.rest.repos.listLanguages,
+};
+
+async function wait(milliseconds) {
+  return new Promise((_resolve) => {
+    if (typeof milliseconds !== "number") {
+      throw new Error("milliseconds not a number");
+    }
+    setTimeout(() => _resolve("done!"), milliseconds);
+  });
+}
+
+async function octokitRequest(request) {
+  console.log(`Running ${request} function`);
+  try {
+    // few functions require different properties
+    let requestProperties = { owner, repo };
+    switch (request) {
+      case "createFork":
+        requestProperties.owner = originalOwner;
+        requestProperties.organization = owner;
+        break;
+    }
+    const response = await octokitFunctions[request](requestProperties);
+    console.log(`Function ${request} finished succesfully`);
+    return response.data;
+  } catch (error) {
+    console.log(`Failed to run ${request}: ${error.message}`);
+  }
+}
+
+async function putRequest(request) {
```

Comment:
Do you _reallllly_ need a function here? probably more future proof but eh

Reply:
Adopted it so its also used by other function